### DISCUSSION TITLE
Using both Ethernet ports on Icicle kit

### DISCUSSION THREAD

**Discussion by andrenemat** [polarfire-soc/polarfire-soc-documentation#211] [How to]
Could you please help with instructions on how to enable both Ethernet ports on the Icicle kit.

**Comment by hughbreslin** [comment]
Hi @andrenemat is this with Linux or bare metal? If its with Linux what build system are you using and what release? They should both be available with Yocto (and I'm pretty sure with Buildroot).

**Reply by andrenemat** [reply]
Trying to use port-j1 in u-boot. By default only port-j2 is recognized, Could not get to use port-j1 in u-boot as well.

**Reply by hughbreslin** [reply]
Ahhh ok - the main change is the u-boot [device tree](https://github.com/polarfire-soc/u-boot/blob/master/arch/riscv/dts/microchip-mpfs-icicle-kit.dts#L149) you can see only mac1 is `status = "okay";` whereas both are present in the [base](https://github.com/polarfire-soc/u-boot/blob/master/arch/riscv/dts/microchip-mpfs.dtsi#L385), but marked as `status = "disabled";`  . I will check in the morning and see if there is any specific reason for this - if not you should be able to enable both in the microchip-mpfs-icicle-kit.dts :)

**Reply by andrenemat** [reply]
Tried in Linux, two ports are available. Could get only one port working at any given time ( could not get two ports working simultaneously).

Tested with Linux version : Linux icicle-kit-es 5.12.19-mpfs-22.03 #1 SMP Tue Apr 19 10:33:07 UTC 2022 riscv64 GNU/Linux


We are finding there is some sort of limitations where only one Ethernet port will work at a time. When we try to use the second port, the first port’s configuration disappears

**Reply by padmaraob** [reply]
> Trying to use port-j1 in u-boot. By default only port-j2 is recognized, Could not get to use port-j1 in u-boot as well.

The U-Boot use only port-j2, the Icicle kit has two Ethernet ports but only one MDIO interface available
between the MAC and the PHY-dual port and i.e connected with Ethernet 1and other port(Ethernet 0) MDIO interface is not connected with PHY in the FPGA design.

In Linux, Ethernet0 MDIO is managed with Enthernet1 MDIO using device tree "phy-handle"

**Reply by andrenemat** [reply]
The dual PHY also has only one MDIO port, you could configure either PHY through the same port by using the appropriate PHY address, I believe, 0 and 1.  Both PHYs are connected to the MSS through SGMII. By the way, you can set up u-boot and Linux to see both PHYs and can use them one at a time, but not both at the same time.
